packages/nameguard-js/src/canonical.ts

```diff
@@ -0,0 +1,59 @@
+import { ens_normalize } from "@adraffy/ens-normalize";
+import { splitGraphemes } from "./graphemes";
+import { graphemeConfusableAnalysis } from "./confusables";
+
+/**
+ * Returns the canonical form of the input label.
+ *
+ * A canonical label is the label where all confusable graphemes are replaced with their canonical form.
+ *
+ * A canonical grapheme is a grapheme that is considered the "base" version of a grapheme, for example, a Latin letter without diacritics.
+ * Some graphemes have no canonical form because of language-specific writing scripts.
+ *
+ * If the input label contains confusable graphemes with no canonical form, the function returns null.
+ *
+ * @param label - The input label.
+ * @returns The canonical label or null if the label contains confusable graphemes with no canonical form.
+ */
+export function getCanonicalLabel(label: string): string | null {
```

Comment:
It's called getCanonicalLabel but it can work on full names as well. The dot is in canonical form. If there's a grapheme that canonicalizes to the label separator dot, then there might be some issues if someone does a second name split (more labels would appear). This sounds quite serious actually. I'll add a note that the input should be a label.

But it seems there could be a special kind of impersonation attack with a fake dot, if such a character exists @lightwalker-eth

Reply:
I added a note that this should be used on single labels but no exception is thrown.